src/main/ruby/core/truffle/regexp_operations.rb

```diff
@@ -19,5 +19,19 @@ def self.match(re, str, pos=0)
       pos = m.character_to_byte_index pos
       re.search_region(str, pos, str.bytesize, true)
     end
+
+    def self.last_match(a_binding)
+      Truffle::KernelOperations.frame_local_variable_get(:$~, a_binding)
+    end
+
+    def self.set_last_match(value, a_binding)
+      unless value.nil? || Truffle::Type.object_kind_of?(value, MatchData)
+        raise TypeError, "Wrong argument type #{value} (expected MatchData)"
+      end
+      Truffle::KernelOperations.frame_local_variable_set(:$~, a_binding, value)
+    end
+
+    Truffle::Graal.always_split(method(:last_match))
+    Truffle::Graal.always_split(method(:set_last_match))
```

Comment:
Please keep method modifiers (private, always_split, etc) just below the method definition.
Otherwise it's hard to see last_match is always split.